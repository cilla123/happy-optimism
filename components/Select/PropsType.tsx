import { BasePickerProps } from '../Picker/PropsType';

export interface BaseSelectProps extends BasePickerProps {
  placeholder?: string;
  displayRender?: (data?: object) => string;
}
