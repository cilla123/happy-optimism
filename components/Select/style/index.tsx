import '../../style/';
import '../../Picker/style';
import './index.scss';
