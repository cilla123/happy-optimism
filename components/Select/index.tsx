import React, { PureComponent } from 'react';
import classnames from 'classnames';
import { BaseSelectProps } from './PropsType';
import Picker from '../Picker';

const getValue = (props, defaultValue?: any) => {
  if (props.value) {
    return [].concat(props.value);
  }

  if (props.defaultValue) {
    return [].concat(props.defaultValue);
  }

  return defaultValue;
};

export interface SelectProps extends BaseSelectProps {
  prefixCls?: string;
  className?: string;
}

export default class Select extends PureComponent<SelectProps, any> {

  static defaultProps = {
    prefixCls: 'za-select',
    valueMember: 'value',
    placeholder: '请选择',
    itemRender: data => data.label,
    displayRender: selected => selected.map(item => item.label),
    onClick: () => {},
  };

  private tempValue;
  private tempObjValue;

  constructor(props) {
    super(props);
    this.state = {
      visible: props.visible || false,
      value: getValue(props, []),
    };
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      value: getValue(nextProps, []),
    });

    if ('visible' in nextProps && nextProps.visible !== this.state.visible) {
      this.setState({
        visible: nextProps.visible,
      });
    }
  }

  onInit = (selected) => {
    const { valueMember } = this.props;
    const firstValue = selected.map(item => item[valueMember!]);

    this.tempValue = this.state.value.length ? firstValue : [] ;
    this.tempObjValue = this.state.value.length ? selected : [];

    this.setState({
      firstValue,
      firstObjValue: selected,
    });
  }

  handleClick = () => {
    this.toggle();
  }

  toggle() {
    if (this.props.disabled) {
      return;
    }

    this.setState({
      visible: !this.state.visible,
    });
  }

  onChange = (selected) => {
    const { onChange } = this.props;
    if (typeof onChange === 'function') {
      onChange(selected);
    }
  }

  onOk = (selected) => {
    this.toggle();
    const { onOk, valueMember } = this.props;
    this.setState({
      value: selected.map(item => item[valueMember!]),
      objValue: selected,
    });
    this.tempValue = selected.map(item => item[valueMember!]);
    this.tempObjValue = selected;
    if (typeof onOk === 'function') {
      onOk(selected);
    }
  }

  // 点击取消
  onCancel = () => {
    const { onCancel } = this.props;
    this.toggle();
    this.setState({
      value: this.tempValue,
      objValue: this.tempObjValue,
    });
    if (typeof onCancel === 'function') {
      onCancel();
    }
  }

  render() {
    const { prefixCls, placeholder, className, disabled, displayRender, ...others } = this.props;
    const { visible, value, objValue, firstObjValue } = this.state;
    const cls = classnames(`${prefixCls}`, className);

    const inputCls = classnames(`${prefixCls}-input`, {
      [`${prefixCls}-placeholder`]: value.length === 0,
      [`${prefixCls}-disabled`]: !!disabled,
    });

    return (
      <div className={cls} onClick={this.handleClick}>
        <div className={inputCls}>
          {value.length > 0 && displayRender!(objValue || firstObjValue || []) || placeholder}
        </div>
        <Picker
          {...others}
          visible={visible}
          value={value}
          firstObjValue={firstObjValue}
          onInit={this.onInit}
          onOk={this.onOk}
          onChange={this.onChange}
          onCancel={this.onCancel}
        />
      </div>
    );
  }
}
