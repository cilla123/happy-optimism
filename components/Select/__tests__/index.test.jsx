import React from 'react';
import { render, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Select from '../index';

describe('Select', () => {
  it('Select', () => {
    const wrapper = mount(
      <Select
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render defaultValue correctly ', () => {
    const wrapper = mount(
      <Select
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        defaultValue="2"
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('Cascader Select', () => {
    // jest.useFakeTimers();
    const wrapper = mount(
      <Select
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.setProps({ value: ['1', '12'] });
    // jest.runAllTimers();
    wrapper.unmount();
  });

  it('Cascader Select init value', () => {
    // jest.useFakeTimers();
    const wrapper = mount(
      <Select
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        value={['1', '12']}
        displayAddon="-"
        />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should trigger onOk when press ok button', () => {
    const onChangeFn = jest.fn();

    const wrapper = mount(
      <Select
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        value={['1', '12']}
        onOk={onChangeFn}
        />
    );

    // wrapper.find('.za-picker-submit').simulate('click');
    // expect(onChangeFn).toBeCalled();
  });

  it('should trigger onCancel when press cancel button', () => {
    const onOkFn = jest.fn();

    const wrapper = mount(
      <Select
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        value={['1', '12']}
        displayAddon="-"
        onOk={onOkFn}
        />
    );

    // wrapper.find('.za-picker-cancel').simulate('click');
    // expect(onOkFn).not.toBeCalled();
  });

  it('receive new dataSource', () => {
    const wrapper = shallow(
      <Select
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    wrapper.setProps({
      dataSource: [
        { value: 'a', label: '选项一' },
        { value: 'b', label: '选项二' },
        { value: 'c', label: '选项三' },
      ],
    });
  });

  it('receive new value', () => {
    const wrapper = shallow(
      <Select
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        />
    );
    wrapper.setProps({ value: '1' });
  });

  it('receive new cascader dataSource', () => {
    const wrapper = shallow(
      <Select
        dataSource={[
          {
            value: '1',
            label: '选项一',
            children: [
              { value: '11', label: '选项一' },
              { value: '12', label: '选项二' },
            ],
          },
          {
            value: '2',
            label: '选项一',
            children: [
              { value: '21', label: '选项一' },
              { value: '22', label: '选项二' },
            ],
          },
        ]}
        />
    );

    wrapper.setProps({
      dataSource: [
        {
          value: '3',
          label: '选项一',
          children: [
            { value: '31', label: '选项一' },
            { value: '32', label: '选项二' },
          ],
        },
        {
          value: '4',
          label: '选项一',
          children: [
            { value: '41', label: '选项一' },
            { value: '42', label: '选项二' },
          ],
        },
      ],
    });
  });

  it('should trigger maskClick when click mask', () => {
    const onMaskClick = jest.fn();
    const wrapper = mount(
      <Select
        dataSource={[
          { value: '1', label: '选项一' },
          { value: '2', label: '选项二' },
        ]}
        onMaskClick={onMaskClick}
        />
    );
    // wrapper.find('.za-select').simulate('click');
    // wrapper.find('.za-mask').simulate('click');
    // expect(onMaskClick).toBeCalled();
  });
});
