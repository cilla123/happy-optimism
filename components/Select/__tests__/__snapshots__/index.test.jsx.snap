// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Cascader Select 1`] = `
<Select
  dataSource={
    Array [
      Object {
        "children": Array [
          Object {
            "label": "选项一",
            "value": "11",
          },
          Object {
            "label": "选项二",
            "value": "12",
          },
        ],
        "label": "选项一",
        "value": "1",
      },
      Object {
        "children": Array [
          Object {
            "label": "选项一",
            "value": "21",
          },
          Object {
            "label": "选项二",
            "value": "22",
          },
        ],
        "label": "选项一",
        "value": "2",
      },
    ]
  }
  displayRender={[Function]}
  itemRender={[Function]}
  onClick={[Function]}
  placeholder="请选择"
  prefixCls="za-select"
  valueMember="value"
>
  <div
    className="za-select"
    onClick={[Function]}
  >
    <div
      className="za-select-input za-select-placeholder"
    >
      请选择
    </div>
    <Picker
      cancelText="取消"
      dataSource={
        Array [
          Object {
            "children": Array [
              Object {
                "label": "选项一",
                "value": "11",
              },
              Object {
                "label": "选项二",
                "value": "12",
              },
            ],
            "label": "选项一",
            "value": "1",
          },
          Object {
            "children": Array [
              Object {
                "label": "选项一",
                "value": "21",
              },
              Object {
                "label": "选项二",
                "value": "22",
              },
            ],
            "label": "选项一",
            "value": "2",
          },
        ]
      }
      firstObjValue={
        Array [
          Object {
            "label": "选项一",
            "value": "1",
          },
          Object {
            "label": "选项一",
            "value": "11",
          },
        ]
      }
      itemRender={[Function]}
      okText="确定"
      onCancel={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onInit={[Function]}
      onOk={[Function]}
      prefixCls="za-picker"
      title="请选择"
      value={Array []}
      valueMember="value"
      visible={false}
    >
      <div
        className="za-picker"
      >
        <Popup
          animationDuration={200}
          autoClose={false}
          direction="bottom"
          mask={true}
          maskType="normal"
          onMaskClick={[Function]}
          prefixCls="za-popup"
          stayTime={3000}
          visible={false}
        >
          <div>
            <Portal
              getContainer={[Function]}
            >
              <div
                className="za-popup za-popup-hidden"
              >
                <div
                  className="za-popup-wrapper za-popup-wrapper-bottom"
                  style={
                    Object {
                      "WebkitTransitionDuration": "200ms",
                      "transitionDuration": "200ms",
                    }
                  }
                >
                  <div
                    className="za-picker-wrapper"
                  >
                    <div
                      className="za-picker-header"
                    >
                      <div
                        className="za-picker-cancel"
                        onClick={[Function]}
                      >
                        取消
                      </div>
                      <div
                        className="za-picker-title"
                      >
                        请选择
                      </div>
                      <div
                        className="za-picker-submit"
                        onClick={[Function]}
                      >
                        确定
                      </div>
                    </div>
                    <PickerView
                      cols={Infinity}
                      dataSource={
                        Array [
                          Object {
                            "children": Array [
                              Object {
                                "label": "选项一",
                                "value": "11",
                              },
                              Object {
                                "label": "选项二",
                                "value": "12",
                              },
                            ],
                            "label": "选项一",
                            "value": "1",
                          },
                          Object {
                            "children": Array [
                              Object {
                                "label": "选项一",
                                "value": "21",
                              },
                              Object {
                                "label": "选项二",
                                "value": "22",
                              },
                            ],
                            "label": "选项一",
                            "value": "2",
                          },
                        ]
                      }
                      disabled={false}
                      firstObjValue={
                        Array [
                          Object {
                            "label": "选项一",
                            "value": "1",
                          },
                          Object {
                            "label": "选项一",
                            "value": "11",
                          },
                        ]
                      }
                      itemRender={[Function]}
                      onCancel={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onInit={[Function]}
                      onOk={[Function]}
                      onTransition={[Function]}
                      prefixCls="za-picker"
                      value={Array []}
                      valueMember="value"
                      visible={false}
                    >
                      <div
                        className="za-picker-panel"
                      >
                        <div
                          className="za-picker-mask-top"
                        />
                        <div
                          className="za-picker-view"
                        >
                          <Wheel
                            dataSource={
                              Array [
                                Object {
                                  "label": "选项一",
                                  "value": "1",
                                },
                                Object {
                                  "label": "选项一",
                                  "value": "2",
                                },
                              ]
                            }
                            disabled={false}
                            itemRender={[Function]}
                            key="0"
                            onChange={[Function]}
                            onTransition={[Function]}
                            prefixCls="za-wheel"
                            value="1"
                            valueMember="value"
                          >
                            <div
                              className="za-wheel"
                            >
                              <div
                                className="za-wheel-content"
                              >
                                <div
                                  className="za-wheel-item za-wheel-item-selected"
                                  key="0"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-wheel-item"
                                  key="1"
                                >
                                  选项一
                                </div>
                              </div>
                            </div>
                          </Wheel>
                          <Wheel
                            dataSource={
                              Array [
                                Object {
                                  "label": "选项一",
                                  "value": "11",
                                },
                                Object {
                                  "label": "选项二",
                                  "value": "12",
                                },
                              ]
                            }
                            disabled={false}
                            itemRender={[Function]}
                            key="1"
                            onChange={[Function]}
                            onTransition={[Function]}
                            prefixCls="za-wheel"
                            value="11"
                            valueMember="value"
                          >
                            <div
                              className="za-wheel"
                            >
                              <div
                                className="za-wheel-content"
                              >
                                <div
                                  className="za-wheel-item za-wheel-item-selected"
                                  key="0"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-wheel-item"
                                  key="1"
                                >
                                  选项二
                                </div>
                              </div>
                            </div>
                          </Wheel>
                        </div>
                        <div
                          className="za-picker-mask-bottom"
                        />
                      </div>
                    </PickerView>
                  </div>
                </div>
                <Mask
                  className="fade-leave"
                  onClose={[Function]}
                  prefixCls="za-mask"
                  style={
                    Object {
                      "WebkitAnimationDuration": "200ms",
                      "animationDuration": "200ms",
                    }
                  }
                  type="normal"
                  visible={false}
                />
              </div>
            </Portal>
          </div>
        </Popup>
      </div>
    </Picker>
  </div>
</Select>
`;

exports[`Select Cascader Select init value 1`] = `
<Select
  dataSource={
    Array [
      Object {
        "children": Array [
          Object {
            "label": "选项一",
            "value": "11",
          },
          Object {
            "label": "选项二",
            "value": "12",
          },
        ],
        "label": "选项一",
        "value": "1",
      },
      Object {
        "children": Array [
          Object {
            "label": "选项一",
            "value": "21",
          },
          Object {
            "label": "选项二",
            "value": "22",
          },
        ],
        "label": "选项一",
        "value": "2",
      },
    ]
  }
  displayAddon="-"
  displayRender={[Function]}
  itemRender={[Function]}
  onClick={[Function]}
  placeholder="请选择"
  prefixCls="za-select"
  value={
    Array [
      "1",
      "12",
    ]
  }
  valueMember="value"
>
  <div
    className="za-select"
    onClick={[Function]}
  >
    <div
      className="za-select-input"
    >
      选项一
      选项二
    </div>
    <Picker
      cancelText="取消"
      dataSource={
        Array [
          Object {
            "children": Array [
              Object {
                "label": "选项一",
                "value": "11",
              },
              Object {
                "label": "选项二",
                "value": "12",
              },
            ],
            "label": "选项一",
            "value": "1",
          },
          Object {
            "children": Array [
              Object {
                "label": "选项一",
                "value": "21",
              },
              Object {
                "label": "选项二",
                "value": "22",
              },
            ],
            "label": "选项一",
            "value": "2",
          },
        ]
      }
      displayAddon="-"
      firstObjValue={
        Array [
          Object {
            "label": "选项一",
            "value": "1",
          },
          Object {
            "label": "选项二",
            "value": "12",
          },
        ]
      }
      itemRender={[Function]}
      okText="确定"
      onCancel={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onInit={[Function]}
      onOk={[Function]}
      prefixCls="za-picker"
      title="请选择"
      value={
        Array [
          "1",
          "12",
        ]
      }
      valueMember="value"
      visible={false}
    >
      <div
        className="za-picker"
      >
        <Popup
          animationDuration={200}
          autoClose={false}
          direction="bottom"
          mask={true}
          maskType="normal"
          onMaskClick={[Function]}
          prefixCls="za-popup"
          stayTime={3000}
          visible={false}
        >
          <div>
            <Portal
              getContainer={[Function]}
            >
              <div
                className="za-popup za-popup-hidden"
              >
                <div
                  className="za-popup-wrapper za-popup-wrapper-bottom"
                  style={
                    Object {
                      "WebkitTransitionDuration": "200ms",
                      "transitionDuration": "200ms",
                    }
                  }
                >
                  <div
                    className="za-picker-wrapper"
                  >
                    <div
                      className="za-picker-header"
                    >
                      <div
                        className="za-picker-cancel"
                        onClick={[Function]}
                      >
                        取消
                      </div>
                      <div
                        className="za-picker-title"
                      >
                        请选择
                      </div>
                      <div
                        className="za-picker-submit"
                        onClick={[Function]}
                      >
                        确定
                      </div>
                    </div>
                    <PickerView
                      cols={Infinity}
                      dataSource={
                        Array [
                          Object {
                            "children": Array [
                              Object {
                                "label": "选项一",
                                "value": "11",
                              },
                              Object {
                                "label": "选项二",
                                "value": "12",
                              },
                            ],
                            "label": "选项一",
                            "value": "1",
                          },
                          Object {
                            "children": Array [
                              Object {
                                "label": "选项一",
                                "value": "21",
                              },
                              Object {
                                "label": "选项二",
                                "value": "22",
                              },
                            ],
                            "label": "选项一",
                            "value": "2",
                          },
                        ]
                      }
                      disabled={false}
                      displayAddon="-"
                      firstObjValue={
                        Array [
                          Object {
                            "label": "选项一",
                            "value": "1",
                          },
                          Object {
                            "label": "选项二",
                            "value": "12",
                          },
                        ]
                      }
                      itemRender={[Function]}
                      onCancel={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onInit={[Function]}
                      onOk={[Function]}
                      onTransition={[Function]}
                      prefixCls="za-picker"
                      value={
                        Array [
                          "1",
                          "12",
                        ]
                      }
                      valueMember="value"
                      visible={false}
                    >
                      <div
                        className="za-picker-panel"
                      >
                        <div
                          className="za-picker-mask-top"
                        />
                        <div
                          className="za-picker-view"
                        >
                          <Wheel
                            dataSource={
                              Array [
                                Object {
                                  "label": "选项一",
                                  "value": "1",
                                },
                                Object {
                                  "label": "选项一",
                                  "value": "2",
                                },
                              ]
                            }
                            disabled={false}
                            itemRender={[Function]}
                            key="0"
                            onChange={[Function]}
                            onTransition={[Function]}
                            prefixCls="za-wheel"
                            value="1"
                            valueMember="value"
                          >
                            <div
                              className="za-wheel"
                            >
                              <div
                                className="za-wheel-content"
                              >
                                <div
                                  className="za-wheel-item za-wheel-item-selected"
                                  key="0"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-wheel-item"
                                  key="1"
                                >
                                  选项一
                                </div>
                              </div>
                            </div>
                          </Wheel>
                          <Wheel
                            dataSource={
                              Array [
                                Object {
                                  "label": "选项一",
                                  "value": "11",
                                },
                                Object {
                                  "label": "选项二",
                                  "value": "12",
                                },
                              ]
                            }
                            disabled={false}
                            itemRender={[Function]}
                            key="1"
                            onChange={[Function]}
                            onTransition={[Function]}
                            prefixCls="za-wheel"
                            value="12"
                            valueMember="value"
                          >
                            <div
                              className="za-wheel"
                            >
                              <div
                                className="za-wheel-content"
                              >
                                <div
                                  className="za-wheel-item"
                                  key="0"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-wheel-item za-wheel-item-selected"
                                  key="1"
                                >
                                  选项二
                                </div>
                              </div>
                            </div>
                          </Wheel>
                        </div>
                        <div
                          className="za-picker-mask-bottom"
                        />
                      </div>
                    </PickerView>
                  </div>
                </div>
                <Mask
                  className="fade-leave"
                  onClose={[Function]}
                  prefixCls="za-mask"
                  style={
                    Object {
                      "WebkitAnimationDuration": "200ms",
                      "animationDuration": "200ms",
                    }
                  }
                  type="normal"
                  visible={false}
                />
              </div>
            </Portal>
          </div>
        </Popup>
      </div>
    </Picker>
  </div>
</Select>
`;

exports[`Select Select 1`] = `
<Select
  dataSource={
    Array [
      Object {
        "label": "选项一",
        "value": "1",
      },
      Object {
        "label": "选项二",
        "value": "2",
      },
    ]
  }
  displayRender={[Function]}
  itemRender={[Function]}
  onClick={[Function]}
  placeholder="请选择"
  prefixCls="za-select"
  valueMember="value"
>
  <div
    className="za-select"
    onClick={[Function]}
  >
    <div
      className="za-select-input za-select-placeholder"
    >
      请选择
    </div>
    <Picker
      cancelText="取消"
      dataSource={
        Array [
          Object {
            "label": "选项一",
            "value": "1",
          },
          Object {
            "label": "选项二",
            "value": "2",
          },
        ]
      }
      firstObjValue={
        Array [
          Object {
            "label": "选项一",
            "value": "1",
          },
        ]
      }
      itemRender={[Function]}
      okText="确定"
      onCancel={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onInit={[Function]}
      onOk={[Function]}
      prefixCls="za-picker"
      title="请选择"
      value={Array []}
      valueMember="value"
      visible={false}
    >
      <div
        className="za-picker"
      >
        <Popup
          animationDuration={200}
          autoClose={false}
          direction="bottom"
          mask={true}
          maskType="normal"
          onMaskClick={[Function]}
          prefixCls="za-popup"
          stayTime={3000}
          visible={false}
        >
          <div>
            <Portal
              getContainer={[Function]}
            >
              <div
                className="za-popup za-popup-hidden"
              >
                <div
                  className="za-popup-wrapper za-popup-wrapper-bottom"
                  style={
                    Object {
                      "WebkitTransitionDuration": "200ms",
                      "transitionDuration": "200ms",
                    }
                  }
                >
                  <div
                    className="za-picker-wrapper"
                  >
                    <div
                      className="za-picker-header"
                    >
                      <div
                        className="za-picker-cancel"
                        onClick={[Function]}
                      >
                        取消
                      </div>
                      <div
                        className="za-picker-title"
                      >
                        请选择
                      </div>
                      <div
                        className="za-picker-submit"
                        onClick={[Function]}
                      >
                        确定
                      </div>
                    </div>
                    <PickerView
                      cols={Infinity}
                      dataSource={
                        Array [
                          Object {
                            "label": "选项一",
                            "value": "1",
                          },
                          Object {
                            "label": "选项二",
                            "value": "2",
                          },
                        ]
                      }
                      disabled={false}
                      firstObjValue={
                        Array [
                          Object {
                            "label": "选项一",
                            "value": "1",
                          },
                        ]
                      }
                      itemRender={[Function]}
                      onCancel={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onInit={[Function]}
                      onOk={[Function]}
                      onTransition={[Function]}
                      prefixCls="za-picker"
                      value={Array []}
                      valueMember="value"
                      visible={false}
                    >
                      <div
                        className="za-picker-panel"
                      >
                        <div
                          className="za-picker-mask-top"
                        />
                        <div
                          className="za-picker-view"
                        >
                          <Wheel
                            dataSource={
                              Array [
                                Object {
                                  "label": "选项一",
                                  "value": "1",
                                },
                                Object {
                                  "label": "选项二",
                                  "value": "2",
                                },
                              ]
                            }
                            disabled={false}
                            itemRender={[Function]}
                            key="0"
                            onChange={[Function]}
                            onTransition={[Function]}
                            prefixCls="za-wheel"
                            value="1"
                            valueMember="value"
                          >
                            <div
                              className="za-wheel"
                            >
                              <div
                                className="za-wheel-content"
                              >
                                <div
                                  className="za-wheel-item za-wheel-item-selected"
                                  key="0"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-wheel-item"
                                  key="1"
                                >
                                  选项二
                                </div>
                              </div>
                            </div>
                          </Wheel>
                        </div>
                        <div
                          className="za-picker-mask-bottom"
                        />
                      </div>
                    </PickerView>
                  </div>
                </div>
                <Mask
                  className="fade-leave"
                  onClose={[Function]}
                  prefixCls="za-mask"
                  style={
                    Object {
                      "WebkitAnimationDuration": "200ms",
                      "animationDuration": "200ms",
                    }
                  }
                  type="normal"
                  visible={false}
                />
              </div>
            </Portal>
          </div>
        </Popup>
      </div>
    </Picker>
  </div>
</Select>
`;

exports[`Select render defaultValue correctly  1`] = `
<Select
  dataSource={
    Array [
      Object {
        "label": "选项一",
        "value": "1",
      },
      Object {
        "label": "选项二",
        "value": "2",
      },
    ]
  }
  defaultValue="2"
  displayRender={[Function]}
  itemRender={[Function]}
  onClick={[Function]}
  placeholder="请选择"
  prefixCls="za-select"
  valueMember="value"
>
  <div
    className="za-select"
    onClick={[Function]}
  >
    <div
      className="za-select-input"
    >
      选项二
    </div>
    <Picker
      cancelText="取消"
      dataSource={
        Array [
          Object {
            "label": "选项一",
            "value": "1",
          },
          Object {
            "label": "选项二",
            "value": "2",
          },
        ]
      }
      defaultValue="2"
      firstObjValue={
        Array [
          Object {
            "label": "选项二",
            "value": "2",
          },
        ]
      }
      itemRender={[Function]}
      okText="确定"
      onCancel={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onInit={[Function]}
      onOk={[Function]}
      prefixCls="za-picker"
      title="请选择"
      value={
        Array [
          "2",
        ]
      }
      valueMember="value"
      visible={false}
    >
      <div
        className="za-picker"
      >
        <Popup
          animationDuration={200}
          autoClose={false}
          direction="bottom"
          mask={true}
          maskType="normal"
          onMaskClick={[Function]}
          prefixCls="za-popup"
          stayTime={3000}
          visible={false}
        >
          <div>
            <Portal
              getContainer={[Function]}
            >
              <div
                className="za-popup za-popup-hidden"
              >
                <div
                  className="za-popup-wrapper za-popup-wrapper-bottom"
                  style={
                    Object {
                      "WebkitTransitionDuration": "200ms",
                      "transitionDuration": "200ms",
                    }
                  }
                >
                  <div
                    className="za-picker-wrapper"
                  >
                    <div
                      className="za-picker-header"
                    >
                      <div
                        className="za-picker-cancel"
                        onClick={[Function]}
                      >
                        取消
                      </div>
                      <div
                        className="za-picker-title"
                      >
                        请选择
                      </div>
                      <div
                        className="za-picker-submit"
                        onClick={[Function]}
                      >
                        确定
                      </div>
                    </div>
                    <PickerView
                      cols={Infinity}
                      dataSource={
                        Array [
                          Object {
                            "label": "选项一",
                            "value": "1",
                          },
                          Object {
                            "label": "选项二",
                            "value": "2",
                          },
                        ]
                      }
                      defaultValue="2"
                      disabled={false}
                      firstObjValue={
                        Array [
                          Object {
                            "label": "选项二",
                            "value": "2",
                          },
                        ]
                      }
                      itemRender={[Function]}
                      onCancel={[Function]}
                      onChange={[Function]}
                      onClick={[Function]}
                      onInit={[Function]}
                      onOk={[Function]}
                      onTransition={[Function]}
                      prefixCls="za-picker"
                      value={
                        Array [
                          "2",
                        ]
                      }
                      valueMember="value"
                      visible={false}
                    >
                      <div
                        className="za-picker-panel"
                      >
                        <div
                          className="za-picker-mask-top"
                        />
                        <div
                          className="za-picker-view"
                        >
                          <Wheel
                            dataSource={
                              Array [
                                Object {
                                  "label": "选项一",
                                  "value": "1",
                                },
                                Object {
                                  "label": "选项二",
                                  "value": "2",
                                },
                              ]
                            }
                            disabled={false}
                            itemRender={[Function]}
                            key="0"
                            onChange={[Function]}
                            onTransition={[Function]}
                            prefixCls="za-wheel"
                            value="2"
                            valueMember="value"
                          >
                            <div
                              className="za-wheel"
                            >
                              <div
                                className="za-wheel-content"
                              >
                                <div
                                  className="za-wheel-item"
                                  key="0"
                                >
                                  选项一
                                </div>
                                <div
                                  className="za-wheel-item za-wheel-item-selected"
                                  key="1"
                                >
                                  选项二
                                </div>
                              </div>
                            </div>
                          </Wheel>
                        </div>
                        <div
                          className="za-picker-mask-bottom"
                        />
                      </div>
                    </PickerView>
                  </div>
                </div>
                <Mask
                  className="fade-leave"
                  onClose={[Function]}
                  prefixCls="za-mask"
                  style={
                    Object {
                      "WebkitAnimationDuration": "200ms",
                      "animationDuration": "200ms",
                    }
                  }
                  type="normal"
                  visible={false}
                />
              </div>
            </Portal>
          </div>
        </Popup>
      </div>
    </Picker>
  </div>
</Select>
`;
